toolkit/tools/internal/testutils/testrpms/download-test-rpms.sh

```diff
@@ -22,11 +22,58 @@ REPO_NO_KEY_FILE="$DOWNLOADER_RPMS_DIRS/rpms-$IMAGE_VERSION-nokey.repo"
 
 mkdir -p "$OUT_DIR"
 
+# Determine which Dockerfile to use and extract package list based on the base image
+TESTDATA_DIR="$SCRIPT_DIR/../../../../tools/pkg/imagecreatorlib/testdata"
+DOCKERFILE_NAME="Dockerfile"
+PACKAGE_LIST=""
+
+if [[ "$CONTAINER_IMAGE" == *"fedora"* ]]; then
+  DOCKERFILE_NAME="Dockerfile.fedora"
+  echo "Using Fedora-specific Dockerfile"
+  # Extract packages from fedora.yaml
+  if command -v python3 >/dev/null 2>&1; then
+    PACKAGE_LIST=$(python3 -c "
+import yaml
```

Comment:
This Python code is almost identical, except for the name of the file. Move this code to a generic Python script and use it for both distros.